nncf/quantization/advanced_parameters.py

```diff
@@ -234,7 +234,7 @@ class AdvancedQuantizationParameters:
     # General parameters
     overflow_fix: OverflowFix = None
     quantize_outputs: bool = False
-    inplace_statistics: bool = True
+    inplace_statistics: bool = False
```

Comment:
Please keep default `inplace_statistics` value false. All backends except OpenVINO ignore this parameter, it is safe to have such a default value

Reply:
I'm sorry if I misunderstood this, since I already changed it to False, do you mean True?

Reply:
Since the param `inplace` is passed now for all backends in `_get_statistic_collector`. Keeping the default value to True causes an error in some tests for backends other than Openvino.

For example FX `test_bias_correction.py`:

![image](https://github.com/user-attachments/assets/7a841224-9241-4701-8241-8f9f3306db17)

When I investigated, I realized that the value of `inplace` was True when passed [here](https://github.com/openvinotoolkit/nncf/pull/3132/files#diff-439982cdce21bd7deb841ee26d4601401f3ca51bd4af3fbe13b18a4d7407da2bR495) for all backends causing the error since some of the reducers [here](https://github.com/openvinotoolkit/nncf/blob/develop/nncf/experimental/common/tensor_statistics/collectors.py#L410-L541) do not implement `get_inplace_fn`. 

I tried changing [this](https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/algorithms/min_max/algorithm.py#L146) to False as default but since the `inplace_statistics` value in [advanced_parameters.py](https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/advanced_parameters.py#L237) is passed to MinMaxQuantization in [create_ptq_pipeline](https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/algorithms/post_training/pipeline.py#L118) it updates [this value]((https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/algorithms/min_max/algorithm.py#L146)) to True as well, being the source of the error.

Reply:
All redusers have a default implementation of `get_inplace_fn`, please check this https://github.com/openvinotoolkit/nncf/blob/develop/nncf/experimental/common/tensor_statistics/collectors.py#L73-L79

Reply:
Yea, it returns `None` though, unlike OV.

Reply:
Could you please send a trace back of the errors you have? From my perspective, `get_inplace_fn` could not cause this errors because it is being called only in openvino code (only one real use here https://github.com/openvinotoolkit/nncf/blob/develop/nncf/openvino/statistics/aggregator.py#L70)

Reply:
Sure.

![image](https://github.com/user-attachments/assets/768facaf-91f3-4f3c-be8c-9b6c75210cef)
![image](https://github.com/user-attachments/assets/268b4934-6136-4857-a47d-ed2df5780971)

Reply:
The problem appears here: https://github.com/openvinotoolkit/nncf/blob/develop/nncf/experimental/common/tensor_statistics/collectors.py#L85-L88 reducer just returns given input value all the time. Let's mimic previous logic and introduce `supports_inplace_statistics` method to `MinMaxAlgoBackend`

Reply:
> The problem appears here: https://github.com/openvinotoolkit/nncf/blob/develop/nncf/experimental/common/tensor_statistics/collectors.py#L85-L88 reducer just returns given input value all the time. 

Oh yes, I remember coming across this. I guess I did not give a clear response earlier on the cause of the error because I traced back and investigated way before and forgot a few things. Should've noted down :) Thank you for clearing up.

> Let's mimic previous logic and introduce `supports_inplace_statistics` method to `MinMaxAlgoBackend`

Alright 👍.